File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
        vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
     int maxCandies = *max_element(candies.begin(), candies.end());
        
        // Step 2: Initialize the result vector
        vector<bool> result;
        
        // Step 3: Check if each kid can have the maximum candies with extraCandies
        for (int i = 0; i < candies.size(); i++) {
            // If the current kid's candies plus extraCandies are at least maxCandies, push tr
            if (candies[i] + extraCandies >= maxCandies) {
                result.push_back(true);
            } else {
                result.push_back(false);
            }
        }

        return result;
    }
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] c, int e) {
        ArrayList<Boolean>list = new ArrayList<>();
        int max=c[0];
        for(int i=1;i<c.length;i++){
            max=Math.max(max,c[i]);
        }
        for(int i=0;i<c.length;i++){
            if(c[i]+e>=max)list.add(true);
            else list.add(false);
        }
        return list;
    }
}